ipalib/x509.py

```diff
@@ -543,3 +562,23 @@ def format_datetime(t):
     if t.tzinfo is None:
         t = t.replace(tzinfo=UTC())
     return unicode(t.strftime("%a %b %d %H:%M:%S %Y %Z"))
+
+
+def match_hostname(cert, hostname):
+    match_cert = {}
+
+    match_cert['subject'] = match_subject = []
+    for rdn in cert.subject.rdns:
```

Comment:
What RFC 5280 actually states is that the subject might be empty, not that the subject field is optional.

Empty subject has an empty list of RDNs:
```
>>> from cryptography.hazmat.backends import default_backend
>>> from cryptography.x509 import load_pem_x509_certificate
>>> load_pem_x509_certificate(open('no-subject.crt', 'rb').read(), default_backend()).subject.rdns
[]
```
i.e. no extra check is necessary.